projects/RabbitMQ.Client/client/RentedMemory.cs

```diff
@@ -0,0 +1,88 @@
+﻿// This source code is dual-licensed under the Apache License, version
+// 2.0, and the Mozilla Public License, version 2.0.
+//
+// The APL v2.0:
+//
+//---------------------------------------------------------------------------
+//   Copyright (c) 2007-2020 VMware, Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       https://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//---------------------------------------------------------------------------
+//
+// The MPL v2.0:
+//
+//---------------------------------------------------------------------------
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+//  Copyright (c) 2007-2020 VMware, Inc.  All rights reserved.
+//---------------------------------------------------------------------------
+
+using System;
+
+namespace RabbitMQ.Client
+{
+    internal struct RentedMemory : IDisposable
```

Comment:
Consider using a `MemoryPool<T>` that rents a disposable `IMemoryOwner<T>` or extending `MemoryManager<T>`.

Reply:
FYI it has been brought up also here https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1347#discussion_r1396389040 and then moved into https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1421 but then the description and title got changed

Reply:
MemoryPool isn't supported by `netstandard2.0`, unfortunately. I will open a version 8 issue to look into supported frameworks and will mention this discussion. Thanks.

Reply:
https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1424